### PR TITLE
added spot update function

### PR DESCRIPTION
**Pull request by damnski**
- Updated README
- Added function comments

### CODE REVIEW

**Comment by tenyo** [comment]
Looks good 👍

File: api/handlers_elastigroups.go [hunk 1]
```diff
@@ -79,7 +79,44 @@ func (s *server) ElastigroupShowHandler(w http.ResponseWriter, r *http.Request)
...
-// ElastgroupCreateHandler handles creating an elastigroup in SpotInst
+// ElastigroupUpdateHandler handles updating an elastigroup in SpotInst
+func (s *server) ElastigroupUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	esService, ok := s.elastigroupServices[account]
+	if !ok {
+		log.Errorf("account not found: %s", account)
+		return
```
Comment (fishnix) [inline api/handlers_elastigroups.go:91]: this needs to set a status code before it returns

File: api/handlers_elastigroups.go [hunk 2]
```diff
@@ -79,7 +79,44 @@ func (s *server) ElastigroupShowHandler(w http.ResponseWriter, r *http.Request)
...
+	output, err := esService.UpdateAWSElastigroup(r.Context(), &req)
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+
+	j, err := json.Marshal(output)
+	if err != nil {
+		log.Errorf("cannot marshal response (%v) into JSON: %s", output, err)
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (fishnix) [inline api/handlers_elastigroups.go:117]: you can use `handleError()` here too

File: api/handlers_elastigroups.go [hunk 3]
```diff
@@ -79,7 +79,44 @@ func (s *server) ElastigroupShowHandler(w http.ResponseWriter, r *http.Request)
...
+
+	req := aws.Group{}
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		msg := fmt.Sprintf("cannot decode body into update elastigroup input: %s", err)
+		handleError(w, apierror.New(apierror.ErrBadRequest, msg, err))
+		return
+	}
+
+	output, err := esService.UpdateAWSElastigroup(r.Context(), &req)
```
Comment (fishnix) [inline api/handlers_elastigroups.go:107]: we should probably override the group id in the body before we pass it here

**Review by fishnix — APPROVED** [review]
just some minor stuff, looks good otherwise 👍

File: api/handlers_elastigroups.go (hunk 2 above)
Comment (damnski) [reply api/handlers_elastigroups.go:117]: Done

File: api/handlers_elastigroups.go (hunk 3 above)
Comment (damnski) [reply api/handlers_elastigroups.go:107]: complete

File: api/handlers_elastigroups.go (hunk 1 above)
Comment (damnski) [reply api/handlers_elastigroups.go:91]: added StatusNotFound

**Pull request merged**